packages/keyring-api/src/api/transaction.ts

```diff
@@ -0,0 +1,296 @@
+import type { Infer } from '@metamask/superstruct';
+import { array, enums, nullable, number, string } from '@metamask/superstruct';
+import { CaipChainIdStruct } from '@metamask/utils';
+
+import { AssetStruct } from './asset';
+import type { InferEquals } from '../superstruct';
+import { object } from '../superstruct';
+import type { Paginated } from '../utils';
+import { UuidStruct } from '../utils';
+
+/**
+ * This struct represents a participant in a transaction.
+ *
+ * @example
+ * ```ts
+ * {
+ *   address: '0x1234...',
+ *   asset: {
+ *     fungible: true,
+ *     type: 'eip155:1/slip44:60',
+ *     unit: 'ETH',
+ *     amount: '0.01',
+ *   },
+ * },
+ * ```
+ *
+ * @example
+ * ```ts
+ * {
+ *   address: '0x1234...',
+ *   asset: {
+ *     fungible: false,
+ *     id: 'eip155:1/erc721:0x06012c8cf97BEaD5deAe237070F9587f8E7A266d/771769',
+ *   },
+ * },
+ * ```
+ */
+const ParticipantStruct = object({
+  /**
+   * Participant address.
+   */
+  address: string(),
+
+  /**
+   * Asset being transferred.
+   */
+  asset: nullable(AssetStruct),
+});
+
+/**
+ * Fee types.
+ */
+export enum FeeType {
+  /**
+   * Base fee. It is the minimum fee required to include a transaction in the
+   * blockchain.
+   *
+   * For non-confirmed transactions, it must be the maximum base fee. For
+   * confirmed transactions, it must be the actual base fee paid.
+   */
+  Base = 'base',
+
+  /**
+   * Priority fee. It is an optional fee used to prioritize the transaction.
+   *
+   * For non-confirmed transactions, it must be the maximum priority fee. For
+   * confirmed transactions, it must be the actual priority fee paid.
+   */
+  Priority = 'priority',
+}
+
+const FeeStruct = object({
+  /**
+   * Fee type.
+   */
+  type: enums([`${FeeType.Base}`, `${FeeType.Priority}`]),
+
+  /**
+   * Asset used to pay for the fee.
+   */
+  asset: AssetStruct,
+});
+
+/**
+ * Transaction statuses.
+ *
+ * The possible values are:
+ *
+ * - submitted: The transaction has been submitted but is not yet in the
+ * blockchain. For example, it can be in the mempool.
+ *
+ * - unconfirmed: The transaction is in the blockchain but has not been
+ * confirmed yet.
+ *
+ * - confirmed: The transaction has been confirmed.
+ *
+ * - failed: The transaction has failed. For example, it has been reverted.
+ */
+export enum TransactionStatus {
+  Submitted = 'submitted',
+  Unconfirmed = 'unconfirmed',
+  Confirmed = 'confirmed',
+  Failed = 'failed',
+}
+
+/**
+ * Transaction types.
+ */
+export enum TransactionType {
+  Send = 'send',
+  Receive = 'receive',
```

Comment:
Also to keep using the same pattern.

```suggestion
  /**
   * The transaction was originated by the account. If the transaction
   * has a change output that goes back to the same account, it must be tagged
   * as a send transaction.
   */
  Send = 'send',
  
  /**
   * The transaction was received by the account, but originated by
   * another account.
   */
  Receive = 'receive',
```